RD-1252 Align Plugin Name vertically in System Resources

Gelio opened the pull request:
This PR aligns the Plugin Name column cell vertically so the plugin name and the icon are roughly on the same line.

I have noticed the same bug in the Sites widget, so I fixed it too in this PR

I assume the change is small enough for 1 approval to be sufficient

## Plugins widget
Before:
![image](https://user-images.githubusercontent.com/889383/111138388-2780e980-8580-11eb-8719-ec0feb751ad9.png)

After:
![image](https://user-images.githubusercontent.com/889383/111138275-07e9c100-8580-11eb-8c62-6dde906c8255.png)

## Sites widget

Before:
![image](https://user-images.githubusercontent.com/889383/111139460-711e0400-8581-11eb-9bf6-89ea9ebd5b87.png)


After:
![image](https://user-images.githubusercontent.com/889383/111139402-5ea3ca80-8581-11eb-8316-be0bb8720f1b.png)

## Snapshots widget
Before:
![image](https://user-images.githubusercontent.com/889383/111145820-fd7ff500-8588-11eb-93e5-a02bf8848128.png)


After:
![image](https://user-images.githubusercontent.com/889383/111145693-d1647400-8588-11eb-9419-1c8de5fb6bed.png)

## Secret Store Managment widget
Before:
![image](https://user-images.githubusercontent.com/889383/111145782-efca6f80-8588-11eb-9394-61aa44f0e57d.png)


After:
![image](https://user-images.githubusercontent.com/889383/111145734-e04b2680-8588-11eb-849f-538924f65a12.png)


## Approaches tried

I have tried the following approaches, but they did not work well:
1. Adding the class to the `td` itself - the padding was then too small
2. Using `vertical-align: middle` (it's already set)
3. Using `line-height` - this one works quite well, but requires the height to be set, and it would have to be set on a case-by-case basis, and not once

kubama commented on app/styles/style.scss:
```diff
@@ -443,6 +443,12 @@ table.servicesData button.refreshButton {
   width: auto !important;
 }
 
+td .verticallyAlignedCell {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+}
+
```
How about adding a simple wrapper component instead of adding a global css rule?

Gelio replied:
Good idea :slightly_smiling_face: Where would you recommend to put it? `Stage.Common`?

kubama replied:
I think so.

qooban replied:
Agree, good idea.

Gelio replied:
Done :slightly_smiling_face:

qooban reviewed (commented):
I think that Snapshots widget can get the same treatment 🙂 

![image](https://user-images.githubusercontent.com/5202105/111143469-20f57080-8586-11eb-8270-3262d09b38d7.png)

Gelio commented:
> I think that Snapshots widget can get the same treatment :slightly_smiling_face: 

Good point :+1: I've fixed the alignment in the Snapshots table and the Secrets table too. I have updated the PR description with the screenshots

kubama commented on widgets/common/src/VerticallyAlignedCell.tsx:
```diff
@@ -0,0 +1,22 @@
+import styled from 'styled-components';
+
+declare global {
+    namespace Stage {
+        interface Common {
+            VerticallyAlignedCell: typeof VerticallyAlignedCell;
+        }
+    }
+}
+
+const VerticallyAlignedCell = styled.div`
+    td & {
+        display: flex;
+        justify-content: space-between;
+        align-items: center;
+    }
+`;
+
+Stage.defineCommon({
+    name: 'VerticallyAlignedCell',
+    common: VerticallyAlignedCell
+});
```
Revisiting how and where this component is now used I think that an even cleaner solution would be to add a prop to `DataTable.Data` and have it handled internally. Yes, I know it's not in line with my previous comment...

Gelio replied:
That would definitely compose better, but I would not add it to the public API prematurely. The reason is that we have not tested this component with other content (a single component, many components), thus, I would consider this experimental.

I would rather move on to the next bug rather than spend time making this perfect. If there will be cases that the component is widely used, I would recommend extracting it at that point.

What do you think?

kubama replied:
> That would definitely compose better, but I would not add it to the public API prematurely. The reason is that we have not tested this component with other content (a single component, many components), thus, I would consider this experimental.

My idea was to add a prop that would enable this only on demand, by default everything would work the old way. So I can see no risk here.

> I would rather move on to the next bug rather than spend time making this perfect. If there will be cases that the component is widely used, I would recommend extracting it at that point.

There is a ticket for adding `className` support for all cloudify-ui-common components, maybe it could be extended to cover this new additional prop for `DataTable.Data` as well?

Gelio replied:
I have found the issue you mentioned: https://cloudifysource.atlassian.net/browse/RD-1151

While it is somewhat related (both mention the components repo), I find moving `VerticallyAlignedCell` to be somewhat separate. I have created https://cloudifysource.atlassian.net/browse/RD-1701 to move the component to `DataTable.Data` prop and linked it to RD-1151 :slightly_smiling_face:

kubama replied:
Cool, thanks.

qooban commented on widgets/common/src/VerticallyAlignedCell.tsx:
```diff
@@ -0,0 +1,22 @@
+import styled from 'styled-components';
```
Is the `size` stage error because of this one? Probably need to add to `externals` in webpack config?

Gelio replied:
Oh, yeah, right... I forgot about it :slightly_smiling_face: Thanks!

Gelio replied:
`styled-components` is not exposed globally, so I reckon moving the component to `Stage.Shared` would have a smaller impact

qooban replied:
`Stage.Shared` was meant to keep components shared between `app` and `widgets`, so it's not the best places (as the new component is not used in `app`), but as we have plan to remove it within https://cloudifysource.atlassian.net/browse/RD-1701, I'm OK.

Gelio commented:
Looks like the bundle sizes are now OK after moving the component to `Stage.Shared`. 

I'm merging the PR :slightly_smiling_face: Thanks for the reviews
